smart-contracts/contracts/CosmosBridge.sol

```diff
@@ -347,10 +347,10 @@ contract CosmosBridge is CosmosBridgeStorage, Oracle {
         );
       } else {
         // if we are double pegging and already control the token, then we are going to need to get the address on this chain
-        tokenAddress = sourceAddressToDestinationAddress[claimData.tokenAddress];
+        tokenAddress = cosmosDenomToDestinationAddress[claimData.cosmosDenom];
       }
     } else {
-      tokenAddress = claimData.tokenAddress;
+        tokenAddress = claimData.tokenAddress;
```

Comment:
What if its a first time double pegged token, how is it getting created?

Reply:
this else branch for not double peg case.